### PR TITLE
[Zeppelin-661] Add a documentation for Shiro authentication

### PR DESCRIPTION
**Pull request by AhyoungRyu**
### What is this PR for?
About a month ago, Shiro authentication for Zeppelin is merged by #586. Even though we already have [SECURITY-README.md](https://github.com/apache/incubator-zeppelin/blob/master/SECURITY-README.md), many people do not know about the existence of this file. So I wrote a docs based on `SECURITY-README.md` to the Zeppelin documentation website to guide step by step for Zeppelin users. 

### What type of PR is it?
Documentation

### Todos
* [x] - Add shiro authentication docs
* [x] - Add **zeppelin.anonymous.allowed** property in `zeppelin-site.md`
* [x] - Indent **Websocket security** section in `SECURITY-README.md`

### Is there a relevant Jira issue?
[ZEPPELIN-661](https://issues.apache.org/jira/browse/ZEPPELIN-661)

### Screenshots (if appropriate)
screen shot 2016-02-12 at 11 29 29 am
screen shot 2016-02-12 at 11 29 53 am

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: docs/manual/shiroauthentication.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+-->
+{% include JB/setup %}
+
+# Shiro authentication for Apache Zeppelin
+[Apache Shiro](http://shiro.apache.org/) is a powerful and easy-to-use Java security framework that performs authentication, authorization, cryptography, and session management. In this documentation, we will explain step by step how Shiro works for Zeppelin notebook authentication. 
+
+When you connect to Apache Zeppelin, you will be asked to enter your credentials. Once you logged in, then you have access to all notes including other user's notes.
+
+## Security Setup
+You can setup **Zeppelin notebook authentication** in some simple steps.
```
Comment (felixcheung) [inline docs/manual/shiroauthentication.md]: Could you add by default what is the settings Zeppelin would use? (anon)

**Comment by felixcheung** [comment]
looks good @hayssams could you review?

**Comment by AhyoungRyu** [comment]
@felixcheung Thanks for the quick review! I addressed the default status of `anon` and `authcBasic` as you said.

**Comment by Leemoonsoo** [comment]
Looks great. Thanks @AhyoungRyu for the nice document.

**Comment by AhyoungRyu** [comment]
@Leemoonsoo My pleasure : )

**Comment by hayssams** [comment]
LGTM
Great doc. Thanks

**Comment by felixcheung** [comment]
merging if no more comments

**Comment by minahlee** [comment]
@felixcheung Could you also build the docs and deploy them after you merge the PRs which includes docs changes? So the updated docs can be reflected to website. You can refer https://github.com/apache/incubator-zeppelin/blob/master/docs/README.md

**Comment by AhyoungRyu** [comment]
+1 for @minahlee

**Comment by minahlee** [comment]
@felixcheung FYI, I just deployed the latest documentation on website.

**Comment by felixcheung** [comment]
thanks, I was not aware of that process. Could you give me permission to add to https://cwiki.apache.org/confluence/display/ZEPPELIN/Committer+Guide

**Comment by Leemoonsoo** [comment]
@felixcheung Permission granted. Sorry i forgot adding your id to cwiki.

**Pull request closed without merging**